### PR TITLE
OSDOCS-2333: Azure Stack Hub MAPI integration

### PR DESCRIPTION
**Pull request by jeana-redhat**
For [OSDOCS-2333](https://issues.redhat.com/browse/OSDOCS-2333)

Adds content for Azure Stack Hub Machine API integration. Will also need release notes announcement in the enterprise-4.10 branch.

Previews:
- [Creating a machine set on Azure Stack Hub](https://deploy-preview-40879--osdocs.netlify.app/openshift-enterprise/latest/machine_management/creating_machinesets/creating-machineset-azure-stack-hub.html)
- Creating infrastructure machine sets > Creating infrastructure machine sets for production environments > Creating machine sets for different clouds > [Sample YAML for a machine set custom resource on Azure Stack Hub](https://deploy-preview-40879--osdocs.netlify.app/openshift-enterprise/latest/machine_management/creating-infrastructure-machinesets.html#machineset-yaml-azure-stack-hub_creating-infrastructure-machinesets)

### CODE REVIEW

File: machine_management/creating-infrastructure-machinesets.adoc [hunk 1]
```diff
@@ -34,6 +34,10 @@ include::modules/machineset-yaml-azure.adoc[leveloffset=+3]
 
 Machine sets running on Azure support non-guaranteed xref:../machine_management/creating_machinesets/creating-machineset-azure.adoc#machineset-non-guaranteed-instance_creating-machineset-azure[Spot VMs]. You can save on costs by using Spot VMs at a lower price compared to standard VMs on Azure. You can xref:../machine_management/creating_machinesets/creating-machineset-azure.adoc#machineset-creating-non-guaranteed-instance_creating-machineset-azure[configure Spot VMs] by adding `spotVMOptions` to the `MachineSet` YAML file.
 
+include::modules/machineset-yaml-azure-stack-hub.adoc[leveloffset=+3]
+
+Machine sets running on Azure Stack Hub support non-guaranteed xref:../machine_management/creating_machinesets/creating-machineset-azure.adoc#machineset-non-guaranteed-instance_creating-machineset-azure[Spot VMs]. You can save on costs by using Spot VMs at a lower price compared to standard VMs on Azure Stack Hub. You can xref:../machine_management/creating_machinesets/creating-machineset-azure.adoc#machineset-creating-non-guaranteed-instance_creating-machineset-azure[configure Spot VMs] by adding `spotVMOptions` to the `MachineSet` YAML file.
```
Comment (alexander-demichev) [inline machine_management/creating-infrastructure-machinesets.adoc:39]: Spot instances are not supported on ASH at all, they are missing from this platform, is it possible to leave a note about it?

File: machine_management/creating_machinesets/creating-machineset-azure-stack-hub.adoc [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+include::modules/machineset-creating.adoc[leveloffset=+1]
+
+//Machine sets that deploy machines as Spot VMs
+include::modules/machineset-non-guaranteed-instance.adoc[leveloffset=+1]
+
+//Creating Spot VMs by using machine sets
+include::modules/machineset-creating-non-guaranteed-instances.adoc[leveloffset=+1]
+
+//Machine sets that deploy machines on Ephemeral OS disks
+include::modules/machineset-azure-ephemeral-os.adoc[leveloffset=+1]
```
Comment (alexander-demichev) [inline machine_management/creating_machinesets/creating-machineset-azure-stack-hub.adoc:29]: We don't have ephemeral disks support for ASH too

**Comment by jeana-redhat** [comment]
@miyadav this is content to support OCPCLOUD-1155 in 4.10. I saw your comment on the dev ticket, so I am tagging you for QE review. If that's wrong, please pass this along to whomever should see it. Thanks!

File: modules/machineset-yaml-azure-stack-hub.adoc [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+endif::infra[]
+ifdef::infra[]
+          node-role.kubernetes.io/infra: "" <2>
+      taints: <4>
+      - key: node-role.kubernetes.io/infra
+        effect: NoSchedule
+endif::infra[]
+      providerSpec:
+        value:
+          apiVersion: azureproviderconfig.openshift.io/v1beta1
```
Comment (sunzhaohua2) [inline modules/machineset-yaml-azure-stack-hub.adoc:85]: In 4.10 this should be `apiVersion: machine.openshift.io/v1beta1`

**Comment by sunzhaohua2** [comment]
Hi @jeana-redhat, sorry @miyadav was on PTO last two days.  I have one comment, others lgtm. Thank you.

**Review by lpettyjo — COMMENTED** [review]
Otherwise, LGTM!

**Pull request merged**